Index: test/data/breakpoints.right
===================================================================
--- test/data/breakpoints.right	(revision 828)
+++ test/data/breakpoints.right	(working copy)
@@ -16,7 +16,7 @@
 # break 10
 Breakpoint 2 file ./gcd.rb, line 10
 # break 11
-*** Line 11 is not a stopping point in file "gcd.rb".
+Breakpoint 3 file ./gcd.rb, line 11
 # continue
 Breakpoint 1 at gcd.rb:6
 gcd.rb:6
@@ -25,13 +25,14 @@
 --> #0 Object.gcd(a#Fixnum, b#Fixnum) at line gcd.rb:6
     #1 at line gcd.rb:18
 # break Object.gcd
-Breakpoint 3 at Object::gcd
+Breakpoint 4 at Object::gcd
 # info break
 Num Enb What
   1 y   at gcd.rb:6
 	breakpoint already hit 1 time
   2 y   at gcd.rb:10
-  3 y   at Object:gcd
+  3 y   at gcd.rb:11
+  4 y   at Object:gcd
 # continue
 Breakpoint 2 at gcd.rb:10
 gcd.rb:10
@@ -42,7 +43,7 @@
 # info program
 Program stopped. It stopped at a breakpoint.
 # c 10
-Breakpoint 3 at Object:gcd
+Breakpoint 4 at Object:gcd
 gcd.rb:4
 def gcd(a, b)
 # info break
@@ -51,7 +52,8 @@
 	breakpoint already hit 1 time
   2 y   at gcd.rb:10
 	breakpoint already hit 1 time
-  3 y   at Object:gcd
+  3 y   at gcd.rb:11
+  4 y   at Object:gcd
 	breakpoint already hit 1 time
 # break foo
 *** Invalid breakpoint location: foo.
@@ -61,7 +63,8 @@
 	breakpoint already hit 1 time
   2 y   at gcd.rb:10
 	breakpoint already hit 1 time
-  3 y   at Object:gcd
+  3 y   at gcd.rb:11
+  4 y   at Object:gcd
 	breakpoint already hit 1 time
 # disable  1
 # info break
@@ -70,7 +73,8 @@
 	breakpoint already hit 1 time
   2 y   at gcd.rb:10
 	breakpoint already hit 1 time
-  3 y   at Object:gcd
+  3 y   at gcd.rb:11
+  4 y   at Object:gcd
 	breakpoint already hit 1 time
 # delete 1
 # # We should see breakpoint 2 but not 1
@@ -78,7 +82,8 @@
 Num Enb What
   2 y   at gcd.rb:10
 	breakpoint already hit 1 time
-  3 y   at Object:gcd
+  3 y   at gcd.rb:11
+  4 y   at Object:gcd
 	breakpoint already hit 1 time
 # # We should still be able to access 2
 # disable 2
@@ -87,7 +92,7 @@
 # disable
 *** "disable" must be followed "display", "breakpoints" or breakpoint numbers.
 # # We should be able to delete 2
-# delete 2 3
+# delete 2 3 4
 # info break
 No breakpoints.
 # # Should get a message about having no breakpoints.
Index: test/data/break_bad.right
===================================================================
--- test/data/break_bad.right	(revision 828)
+++ test/data/break_bad.right	(working copy)
@@ -18,10 +18,10 @@
 # # Line one isn't a valid stopping point.
 # # It is a comment.
 # break gcd.rb:1
-*** Line 1 is not a stopping point in file "gcd.rb".
+Breakpoint 1 file gcd.rb, line 1
 # # This line is okay
 # break gcd.rb:4
-Breakpoint 1 file gcd.rb, line 4
+Breakpoint 2 file gcd.rb, line 4
 # # No class Foo.
 # break Foo.bar
 *** Unknown class Foo.
Index: test/data/breakpoints.cmd
===================================================================
--- test/data/breakpoints.cmd	(revision 828)
+++ test/data/breakpoints.cmd	(working copy)
@@ -30,7 +30,7 @@
 disable bar
 disable
 # We should be able to delete 2
-delete 2 3
+delete 2 3 4
 info break
 # Should get a message about having no breakpoints.
 disable 1
Index: test/config.private.yaml
===================================================================
--- test/config.private.yaml	(revision 0)
+++ test/config.private.yaml	(revision 0)
@@ -0,0 +1,5 @@
+# either should be on the $PATH or use full path
+ruby: jruby
+
+# possibility to specify interpreter parameters
+ruby_params: -J-Djruby.reflection=true -J-Djruby.compile.mode=OFF
Index: test/helper.rb
===================================================================
--- test/helper.rb	(revision 828)
+++ test/helper.rb	(working copy)
@@ -1,6 +1,7 @@
 # Some common routines used in testing.
 
 require 'fileutils'
+require 'yaml'
 # require 'diff/lcs'
 # require 'diff/lcs/hunk'
 
@@ -21,12 +22,15 @@
     end
     
     ENV['RDEBUG'] = debug_pgm
+
     if old_code
       cmd = "/bin/sh #{File.join('..', 'runner.sh')} #{args} >#{outfile}"
     else
-      cmd = "../rdbg.rb #{args} >#{outfile}"
+      ruby = config_load('ruby', true)
+      params = config_load('ruby_params', true)
+      cmd = "#{"#{ruby} #{params} "}../rdbg.rb #{args} > #{outfile}"
     end
-    # puts cmd
+    #puts "'#{cmd}'"
     output = `#{cmd}`
     
     got_lines     = File.read(outfile).split(/\n/)
@@ -112,6 +116,15 @@
   #   # Handle the last remaining hunk 
   #   output << oldhunk.diff(format) << '\n'
   # end
-
+  
+  # Loads key from the _config_._yaml_ file.
+  def config_load(key, may_be_nil=false, default_value='')
+    conf = File.join('config.private.yaml') # try private first
+    conf = File.join('config.yaml') unless File.exists?(conf)
+    value = YAML.load_file(conf)[key]
+    assert_not_nil(value, "#{key} is set in config.yaml") unless may_be_nil
+    value || default_value
+  end
+  
 end
 
